python/ray/tune/trainable.py

```diff
@@ -241,16 +241,12 @@ def save(self, checkpoint_dir=None):
         checkpoint = self._save(checkpoint_dir)
         saved_as_dict = False
         if isinstance(checkpoint, string_types):
-            if (not checkpoint.startswith(checkpoint_dir)
-                    or checkpoint == checkpoint_dir):
+            if (not (checkpoint.startswith(checkpoint_dir)
+                     or checkpoint == checkpoint_dir)):
```

Comment:
Why change this check? Do we want to allow the user to return the checkpoint dir?

Reply:
Yeah, this is good especially when you need to save different things in different formats - i.e., `tf.Saver` output and metadata into their own independent files.

Each checkpoint_dir is iteration-suffixed anyways.

Reply:
I'd also prefer using directories for the checkpoints. In fact, I was always expecting this to be the case.